frontend/kubecloud/src/components/dashboard/OverviewCard.vue

```diff
@@ -5,7 +5,7 @@
       <p class="dashboard-card-subtitle">Your Mycelium Cloud platform at a glance</p>
     </div>
     <!-- Stats Grid -->
-    <StatsGrid :stats="statsData" />
+    <StatsGrid :key="netBalance" :stats="statsData" />
```

Comment:
I suggest setting the key on the inner component value only, to reduce the extra rerenderings